### PR TITLE
feat(notifications): Create subscription collection

### PR DESCRIPTION
**Pull request by tiagopazhs**
## Required points discussed in the meeting of May-12:
- [x] Finalize the collection.
- [x] Update the collection names as similar as notification.
- [x] Add the behavior to create a subscription when user creates an idea.
- [x] Add test for idea creation and subscription create.
- [x] ~Fix this test ``expect(subscription.attributes.user.id).toBe(user.id)``~ (TBD in during Notification feature integration)
- [x] ~Verify that subscription are not available in API.~ (TBD in during Notification feature integration)

### CODE REVIEW

**Comment by becevka** [comment]
Please add changes from types/generated/contentTypes.d.ts

File: src/api/subscription/content-types/subscription/schema.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "info": {
+    "singularName": "subscription",
+    "pluralName": "subscriptions",
+    "displayName": "Subscription"
+  },
+  "options": {
+    "draftAndPublish": true
+  },
+  "pluginOptions": {},
+  "attributes": {
```
Comment (chungthuang) [inline src/api/subscription/content-types/subscription/schema.json:13]: Do we want to allow user to select which entity type to subscribe https://github.com/dev-launchers/strapiv4/blob/main/src/api/notification/content-types/notification/schema.json#L27?

**Comment by tiagopazhs** [comment]
This PR belongs to https://github.com/dev-launchers/strapiv4/issues/176

File: types/generated/contentTypes.d.ts [hunk 2]
```diff
@@ -666,6 +666,11 @@ export interface PluginUsersPermissionsUser extends Schema.CollectionType {
       'oneToMany',
       'api::comment.comment'
     >;
+    subscriptions: Attribute.Relation<
+      'plugin::users-permissions.user',
+      'oneToMany',
+      'api::subscription.subscription'
+    >;
     createdAt: Attribute.DateTime;
```
Comment (tiagopazhs) [inline types/generated/contentTypes.d.ts:674]: @becevka is this file that you were talking about?
https://github.com/dev-launchers/strapiv4/pull/181#issuecomment-2106120482

**Comment by tiagopazhs** [comment]
> Please add changes from types/generated/contentTypes.d.ts

Done!
You mean, this https://github.com/dev-launchers/strapiv4/pull/181#discussion_r1615252951 ?

What this file is used for?

File: src/api/subscription/content-types/subscription/schema.json (hunk 1 above)
Comment (tiagopazhs) [reply src/api/subscription/content-types/subscription/schema.json:14]: Yeah!

File: types/generated/contentTypes.d.ts (hunk 2 above)
Comment (chungthuang) [reply types/generated/contentTypes.d.ts:674]: I think this is autogenerated content?

**Comment by tiagopazhs** [comment]
So, I am trying to create an idea-card but I am getting this error:

image

My auth:
image

My idea-card permission:
image

Providers:
image

**Comment by tiagopazhs** [comment]
## Required test that we must have to pass when we are running the tests:
Necessary to deal with this situation: the user is present in the post request but we cannot get it to check if it is correct:

`expect(subscription.attributes.user.id).toBe(user.id)`

image

**Comment by chungthuang** [comment]
> ## Required test that we must have to pass when we are running the tests:
> 
> Necessary to deal with this situation: the user is present in the post request but we cannot get it to check if it is correct:
> 
> `expect(subscription.attributes.user.id).toBe(user.id)`
> 
> image

My guess is user is another collection, so you need to pass a query string `?populate=user` to get user data back. https://docs.strapi.io/dev-docs/api/rest/populate-select#population explains why.

**Comment by tiagopazhs** [comment]
> * Fix this test `expect(subscription.attributes.user.id).toBe(user.id)`

Hey @becevka this is something that was already solved in another branch ?

**Comment by becevka** [comment]
> > * Fix this test `expect(subscription.attributes.user.id).toBe(user.id)`
> 
> Hey @becevka this is something that was already solved in another branch ?

Not yet, that would be something separate, maybe next week

**Review by becevka — APPROVED** [review]
Looks good to me

**Pull request merged**